plugins/pulp_rpm/plugins/importers/yum/purge.py

```diff
@@ -263,3 +264,28 @@ def get_remote_units(file_function, tag, process_func):
     finally:
         file_handle.close()
     return remote_named_tuples
+
+
+def remove_unit_duplicate_nevra(unit_key, type_id, repo_id):
```

Comment:
This function name might be more readable as "remove_duplicate_units". The code below seems like it would work with units that don't necessarily have NEVRA, so it may not be that specific to RPMs/SRPMs.